frontend/src/libs/components/note/note.tsx

```diff
@@ -0,0 +1,175 @@
+        selectedJournalEntry: journal.selectedJournalEntry,
+      };
+    },
+  );
+
+  const { isDirty, control } = useAppForm({
+    defaultValues: {
+      title: selectedJournalEntry?.title,
+      text: selectedJournalEntry?.text,
+    },
+    mode: 'onChange',
+  });
+
+  const { field: titleField } = useFormController({ name: 'title', control });
+  const { field: textField } = useFormController({ name: 'text', control });
+  const { value: titleValue, onChange: onTitleChange } = titleField;
+  const { value: textValue, onChange: onTextChange } = textField;
+
+  const dispatch = useAppDispatch();
+  const { id } = useParams();
+
+  const titleReference = useRef<HTMLDivElement | null>(null);
+  const textReference = useRef<HTMLDivElement | null>(null);
+  const cursorPosition = useRef<number | null>(null);
+
+  const handleSaveNote = useCallback(
+    (data: NoteContent) => {
+      if (data.title && data.text && id) {
+        void dispatch(
+          journalActions.updateJournalEntry({
+            id: Number(id),
+            body: {
+              title: sanitizeInput(data.title),
+              text: sanitizeInput(data.text),
+            },
+          }),
+        );
+      }
+    },
+    [dispatch, id],
+  );
+
+  const handleNoteChange = useCallback(
+    (
+      elementReference: React.MutableRefObject<HTMLDivElement | null>,
+      onChange: (value: string) => void,
+    ): void => {
+      if (elementReference.current) {
+        const newValue = elementReference.current.textContent ?? '';
+        onChange(newValue);
+        changeCursorPosition(cursorPosition);
+      }
+    },
+    [],
+  );
+
+  const handleTitleChange = useCallback(() => {
+    handleNoteChange(titleReference, onTitleChange);
+  }, [handleNoteChange, onTitleChange]);
+
+  const handleTextChange = useCallback(() => {
+    handleNoteChange(textReference, onTextChange);
+  }, [handleNoteChange, onTextChange]);
+
+  useEffect(() => {
+    void dispatch(journalActions.getAllJournalEntries()).then(() => {
+      if (id) {
+        void dispatch(journalActions.setSelectedJournalEntry(Number(id)));
+      }
+    });
+  }, [dispatch, id]);
+
+  useEffect(() => {
+    const handleSaveNoteWithDebounce = debounce((data: NoteContent) => {
+      if (id && isDirty) {
+        handleSaveNote(data);
+      }
+    }, NOTE_TIMEOUT);
+
+    handleSaveNoteWithDebounce({ title: titleValue, text: textValue });
+
+    return () => {
+      handleSaveNoteWithDebounce.clear();
+    };
+  }, [titleValue, textValue, isDirty, userId, id, handleSaveNote]);
+
+  useEffect(() => {
+    if (selectedJournalEntry) {
+      onTitleChange(selectedJournalEntry.title);
+      onTextChange(selectedJournalEntry.text);
+    }
+  }, [onTitleChange, onTextChange, selectedJournalEntry, isDirty]);
+
+  useEffect(() => {
+    if (titleReference.current && cursorPosition.current) {
+      getCursorPosition(titleReference.current, cursorPosition);
```

Comment:
why do we need all this logic with getCursorPosition?

Reply:
Well, without this logic div contentEditable sets the cursor at the beginning and adds characters before the first character ( like from right to left, not as we want - from left to right). Also, we cannot put a cursor somewhere in the middle of the text if you want to edit something there.

Reply:
in the example [here](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/contenteditable) it works well, or you mean something else?

Reply:
I am not sure why I have this behavior, but it looks like this. Also there many requests in google with this problem.

https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/102380564/2662d08f-e365-492d-b954-f03862c546b7

Reply:
Should I change it to regular input so as not to have this logic?

Reply:
@Liza-Veis  Also, I found a problem with a cursor. After the page reloads, or first rendering the component I can't type something, because cursor jumps to the beginning instead of being in the end, and it is impossible to type. If I change note for another one, I can type and update it. And I can't figure out how to deal with this problem.